libcnb-data/src/buildpackage.rs

```diff
@@ -0,0 +1,92 @@
+use crate::buildpackage::PlatformOs::Linux;
+use serde::{Deserialize, Serialize};
+
+fn platform_default() -> Platform {
+    Platform::default()
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Buildpackage {
+    pub buildpack: BuildpackageUri,
+    #[serde(default)]
+    pub dependencies: Vec<BuildpackageUri>,
+    #[serde(default = "platform_default")]
+    pub platform: Platform,
+}
```

Comment:
This was confusing to me, as this value is not really the buildpackage URI, but a struct that has the URI. At first I thought that type was introduced as a newtype for extra semantics (which I don't see much of a value here).

The fact that `dependencies` and `buildpack` share the same TOML table structure is more a coincidence than actual fact. Ideally we would have two types here `BuildpackageBuildpack` and `BuildpackageDependency` that both have an `uri` field. Besides the incorrect naming, should upstream ever decides to add extra metadata to either but not the other, the current approach will be a breaking change.

If we're fine with a breaking change, we should resolve the TOML table at parse-time and make `buildpack` just an `Uri` and `dependencies` a `Vec<Uri>`. I'm not a fan of that approach, but it would be fine at this point in time.